internal/restore/pvc_action.go

```diff
@@ -96,6 +96,9 @@ func resetPVCSpec(pvc *corev1api.PersistentVolumeClaim, vsName string) {
 	}
 	pvc.Spec.DataSource = dataSourceRef
 	pvc.Spec.DataSourceRef = dataSourceRef
+
+	// Clean the selector to make the PVC to dynamically allocate PV.
+	pvc.Spec.Selector = nil
```

Comment:
Let's avoid do this in a specific RIA, reason:
During restore, Velero calls RIAs recursively. We cannot guarantee that this `pvc.Spec.Selector` is not added by some former RIA. As a result, if we remove it, the restore may encounter problem.

Let's do this in Velero backup instead at here https://github.com/vmware-tanzu/velero/blob/0945879a8a64cfda0fe55875d6451c1f16cc1385/pkg/backup/backup_pv_action.go#L76C1-L76C1, this would solve the similar problems for all backup types in one shot

Reply:
Got it. Will create a new PR in Velero repository to address this.